src/ser.rs

```diff
@@ -1,9 +1,111 @@
 use std::fmt;
 use std::fmt::{Display, Formatter};
-use bech32::{Bech32, ToBase32, u5};
+use bech32::{ToBase32, u5, WriteBase32, Base32Len};
 
 use ::*;
 
+/// Converts a stream of bytes written to it to base32. On finalization the according padding will
+/// be applied. That means the results of writing two data blocks with one or two `BytesToBase32`
+/// converters will differ.
+struct BytesToBase32<'a, W: WriteBase32 + 'a> {
```

Comment:
I'm really confused why all this code is needed on top of the bech32-provided Base32 bits?

Reply:
Let `b1` and `b2` be some byte slices. The result of `base32(b1) concat base32(b2)` generally isn't the same as `base32(b1 concat b2)`. So either I allocate a new byte vector to concatenate the slices first or I build this on-the-fly converter. Ideally the converter included in rust-bech32 can be rewritten so that it becomes a mere wrapper around this one, but I didn't want to block on this.